### ISSUE TITLE
1.21.3 console error

### ISSUE THREAD

**Issue by makegodhere** [GrimAnticheat/Grim#1848]
### Stack trace

```
[12:56:01 INFO]: [GrimAC] [STDOUT] An error has occurred when running transactions for player: PalladiuM
[12:56:01 WARN]: java.lang.ArrayIndexOutOfBoundsException: Index -2 out of bounds for length 4
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.shaded.com.github.retrooper.packetevents.protocol.world.Direction.getByHorizontalIndex(Direction.java:43)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.utils.latency.CompensatedEntities.addEntity(CompensatedEntities.java:186)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.events.packets.PacketEntityReplication.lambda$addEntity$24(PacketEntityReplication.java:483)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.utils.latency.LatencyUtils.handleNettySyncTransaction(LatencyUtils.java:56)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.player.GrimPlayer.addTransactionResponse(GrimPlayer.java:356)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.events.packets.PacketPingListener.onPacketReceive(PacketPingListener.java:62)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.shaded.com.github.retrooper.packetevents.event.PacketReceiveEvent.call(PacketReceiveEvent.java:44)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.shaded.com.github.retrooper.packetevents.event.EventManager.callEvent(EventManager.java:84)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.shaded.com.github.retrooper.packetevents.util.PacketEventsImplHelper.handleServerBoundPacket(PacketEventsImplHelper.java:101)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.shaded.io.github.retrooper.packetevents.injector.handlers.PacketEventsDecoder.read(PacketEventsDecoder.java:57)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.shaded.io.github.retrooper.packetevents.injector.handlers.PacketEventsDecoder.decode(PacketEventsDecoder.java:64)
[12:56:01 WARN]:        at grimac-2.3.68.jar//ac.grim.grimac.shaded.io.github.retrooper.packetevents.injector.handlers.PacketEventsDecoder.decode(PacketEventsDecoder.java:41)
[12:56:01 WARN]:        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:88)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at ProtocolLib.jar//com.comphenix.protocol.injector.netty.channel.InboundProtocolReader.channelRead(InboundProtocolReader.java:25)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at io.netty.handler.flow.FlowControlHandler.dequeue(FlowControlHandler.java:202)
[12:56:01 WARN]:        at io.netty.handler.flow.FlowControlHandler.channelRead(FlowControlHandler.java:164)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
[12:56:01 WARN]:        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
[12:56:01 WARN]:        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:333)
[12:56:01 WARN]:        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:454)
[12:56:01 WARN]:        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
[12:56:01 WARN]:        at net.minecraft.server.network.ServerConnectionListener$1$1.channelRead(ServerConnectionListener.java:164)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at io.netty.handler.flush.FlushConsolidationHandler.channelRead(FlushConsolidationHandler.java:152)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[12:56:01 WARN]:        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
[12:56:01 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[12:56:01 WARN]:        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
[12:56:01 WARN]:        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800)
[12:56:01 WARN]:        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:509)
[12:56:01 WARN]:        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
[12:56:01 WARN]:        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[12:56:01 WARN]:        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[12:56:01 WARN]:        at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Grim version

Last build https://github.com/GrimAnticheat/Grim/actions/runs/12120821603

### Server version

1.21.3 paper

### Plugins

not matter

**Comment by SamB440** [comment]
Plugins do matter. Reproduce it without plugins.

**Comment by makegodhere** [comment]
> Plugins do matter. Reproduce it without plugins.

On 1.21.1 all be good.
Error apper only on update paper to 1.21.3
Plugins not changing.

**Comment by AoElite** [comment]
@makegodhere should be fixed in latest commit

**Comment by AoElite** [comment]
Fixed by 3f01edba9d22d90c301bbbcb7a8835369d93182c

**Comment by makegodhere** [comment]
@AoElite yep https://github.com/GrimAnticheat/Grim/actions/runs/12144638070 build fixed this error - thx
